egs/wsj0-mix/DeepClustering/model.py

```diff
@@ -32,34 +21,141 @@ def make_model_and_optimizer(conf):
     The main goal of this function is to make reloading for resuming
     and evaluation very simple.
     """
-    enc = fb.Encoder(fb.STFTFB(**conf['filterbank']))
-    masker = ChimeraPP(int(enc.filterbank.n_feats_out/2), 2,
-                       embedding_dim=20, n_layers=2, hidden_size=600, \
-                       dropout=0.5, bidirectional=True, \
-                       log=conf['filterbank']['log'])
-    model = Model(enc, masker)
+    enc, dec = fb.make_enc_dec('stft', **conf['filterbank'])
+    masker = Chimera(enc.n_feats_out // 2,
+                     **conf['masknet'])
+    model = Model(enc, masker, dec)
     optimizer = make_optimizer(model.parameters(), **conf['optim'])
     return model, optimizer
 
-def load_best_model(train_conf, best_model_path):
+
+class Chimera(nn.Module):
+    def __init__(self, in_chan, n_src, rnn_type='lstm', n_layers=2,
+                 hidden_size=600, bidirectional=True, dropout=0.3,
+                 embedding_dim=20, take_log=False):
+        super().__init__()
+        self.input_dim = in_chan
+        self.n_src = n_src
+        self.take_log = take_log
+        # RNN common
+        self.embedding_dim = embedding_dim
+        self.rnn = SingleRNN(rnn_type, in_chan, hidden_size,
+                             n_layers=n_layers, dropout=dropout,
+                             bidirectional=bidirectional)
+        self.dropout = nn.Dropout(dropout)
+        rnn_out_dim = hidden_size * 2 if bidirectional else hidden_size
+        # Mask heads
+        self.mask_layer = nn.Linear(rnn_out_dim, in_chan * self.n_src)
+        self.mask_act = nn.Sigmoid()  # sigmoid or relu or softmax
+        # DC head
+        self.embedding_layer = nn.Linear(rnn_out_dim, in_chan * embedding_dim)
+        self.embedding_act = nn.Tanh()  # sigmoid or tanh
+
+    def forward(self, input_data):
+        batch, _, n_frames = input_data.shape
+        if self.take_log:
+            input_data = torch.log(input_data + EPS)
+        # Common net
+        out = self.rnn(input_data.permute(0, 2, 1))
+        out = self.dropout(out)
+
+        # DC head
+        proj = self.embedding_layer(out)  # batch, time, freq * emb
+        proj = self.embedding_act(proj)
+        proj = proj.view(batch, n_frames, -1, self.embedding_dim).transpose(1, 2)
+        # (batch, freq * frames, emb)
+        proj = proj.reshape(batch, -1, self.embedding_dim)
```

Comment:
The bug was here. Without the transpose, the time bins where not aligned with each other and training was impossible. 
I added a note about it in the DC loss.